src/iter/internal.rs

```diff
@@ -147,8 +148,7 @@ impl Splitter {
     fn thief_id() -> usize {
         // The actual `ID` value is irrelevant.  We're just using its TLS
         // address as a unique thread key, faster than a real thread-id call.
-        thread_local!{ static ID: bool = false }
-        ID.with(|id| id as *const bool as usize)
+        thread_id::get()
```

Comment:
The comment should be updated too, like:

    // The actual `ID` value is irrelevant, as long as it's unique.